vignettes/compareToBaseR.Rmd

```diff
@@ -0,0 +1,683 @@
+str_trim("\n\nString with trailing and leading white space\n\n")
+
+## base
+trimws(" String with trailing and leading white space\t")
+trimws("\n\nString with trailing and leading white space\n\n")
+```
+
+The stringr function `str_squish()` allows for extra whitespace within a string to be trimmed (in contrast to `str_trim()` which removes whitespace at the beginning and/or end of string). In base R, one might take advantage of `gsub()` to accomplish the same effect.
+
+```{r}
+## stringr
+str_squish(" String with trailing, middle,   and leading white space\t")
+str_squish("\n\nString with excess, trailing and leading white space\n\n")
+```
+
+## `str_wrap`
+
+*Wrap strings into nicely formatted paragraphs*
+
+I'm not sure why these outputs don't match. The stringr packages's `str_wrap()` is a wrapper around `stringi::stri_wrap()` which has defaults that are supposed to be compatible with base R's `strwrap()`.
+
+```{r}
+thanks_path <- file.path(R.home("doc"), "THANKS")
+thanks <- str_c(readLines(thanks_path), collapse = "\n")
+thanks <- word(thanks, 1, 3, fixed("\n\n"))
+
+## stringr
+cat(str_wrap(thanks), "\n")
+
+## base
+cat(strwrap(thanks, width = 80)) ## match default width
+```
+
+
+# Mutate strings
+
+## `str_replace`
+
+*Replace matched patterns in a string*
+
+To replace certain patterns within a string, stringr provides the functions `str_replace()` and `str_replace_all()`. The base R equivalents are `sub()` and `gsub()`. Note the difference in default input order again. In stringr the string comes first, followed by the pattern and replacement, while in base R, the pattern and replacment come before the string.
+
+```{r}
+fruits <- c("apple", "banana", "pear", "pinapple")
+
+## stringr
+str_replace(string = fruits, pattern = "[aeiou]", replacement = "-")
+str_replace_all(string = fruits, pattern = "[aeiou]", replacement = "-")
+
+## base
+sub(pattern = "[aeiou]", replacement = "-", x = fruits)
+gsub(pattern = "[aeiou]", replacement = "-", x = fruits)
+```
+
+
+## case
+
+*Convert case of a string*
+
+Both stringr and base R have functions to convert to upper and lower case.  Title case is also provided in stringr.
+
+
+```{r}
+dog <- "The quick brown dog"
+
+## stringr
+str_to_upper(dog)
+str_to_lower(dog)
+str_to_title(dog)
+
+## base
+toupper(dog)
+tolower(dog)
+tools::toTitleCase(dog)
+```
+
+In stringr we can control the locale, while in base R locale distinctions are controlled with global variables.
+
+```{r}
+## stringr
+str_to_upper("i") # English
+str_to_upper("i", "tr") # Turkish
+```
+
+```{r,eval=F}
+## base
+Sys.setlocale(category = "LC_ALL", locale = "Turkish") # not run
+```
+
+# Join and split
+
+## `str_c`
+
+*Join multiple strings into a single string*
+
+To join multiple strings into one, base R uses `paste()` while string_r uses `str_c()` (the "c" stands for concatenate).
+
+```{r}
+## stringr
+str_c("Letter: ", letters)
+
+## base
+paste("Letter: ", letters, sep = "")
```

Comment:
`paste0()` is a fairer comparison here. I don't think there's much benefit to using `str_c` these days.

It might be worth mentioning `str_flatten()` as a function always returns a character vector of length 1.